### PR TITLE
Add new --ignore-words option to ignore 'good' words

### PR DESCRIPTION
**Pull request by thdot**
A new command line option `-I / --ignore-words` is implemented to specify one or multiple files containing _good_ words which codespell will ignore.
This PR is related to issue #66.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
There are style issues, see Travis output

**Comment by thdot** [comment]
I have updated the commit. Any other objections to the PR?

**Comment by Eric89GXL** [comment]
Can you add a test to make sure it works?

**Comment by thdot** [comment]
I have added unit tests for the ` --ignore-words `and `--regex` option. 
It was a little bit tricky because codespell_lib stores state in global variables. I solved this by adding a test fixture (setup and teardown methods) which reload the python module.

File: codespell_lib/tests/test_basic.py [hunk 1]
```diff
@@ -138,24 +149,46 @@ def test_summary():
...
+            cs.main(f.name, '--summary')
         assert_equal(sio[1], '')  # stderr
         assert_true('SUMMARY' in sio[0])
         assert_equal(len(sio[0].split('\n')), 7)
         assert_true('abandonned' in sio[0].split()[-2])
 
 
+@with_setup(reload_codespell_lib, reload_codespell_lib)
+def test_ignore_dictonary():
+    """Test ignore dictonary functionality"""
```
Comment (luzpaz) [inline codespell_lib/tests/test_basic.py]: typo Line 171,172 should be `dictionary`

Comment (thdot) [inline codespell_lib/tests/test_basic.py]: Ups, thank you.

**Comment by thdot** [comment]
@Eric89GXL @lucasdemarchi Ready to merge?

**Comment by Eric89GXL** [comment]
LGTM. @lucasdemarchi do you want to look?

**Comment by luzpaz** [comment]
@thdot what's the format of this file of good words? I have a list that I'd like to exclude:
```
allEdges
Calculater
dum
eiter
iff
indicies
NormalY
nTo
ot
pres
recurrance
substraction
thru
```

**Comment by thdot** [comment]
It's simply one 'good' word per line.

**Comment by luzpaz** [comment]
Note: the whitelist word needs to be identical to the word that is in dictionary.txt discovered this after encountering: https://github.com/lucasdemarchi/codespell/pull/143#issuecomment-334220687

@thdot perhaps the `-I` feature could be case-insensitive some how?

**Pull request merged**